base_layer/core/src/transaction.rs

```diff
@@ -208,6 +213,28 @@ impl TransactionKernel {
         self.lock_height = lock_height;
         self
     }
+
+    pub fn verify_signature(&self) -> Result<(), TransactionError> {
+        if self.excess.is_some() && self.excess_sig.is_some() {
```

Comment:
You can reduce nesting by saying
    `if self.excess.is_none() || self.excess_sig.is_none() {return.... }`

Reply:
Done